### PR TITLE
Partner Funcs + Splupdate

### PR DESCRIPTION
**Pull request by ethteck**

### CODE REVIEW

**Comment by ethteck** [comment]
85%, fruggers

**Comment by BowserSlug** [comment]
🚀🚀🚀🚀🚀🚀 This PR matches 6 functions (+0.21%) on `us`.

File: src/battle/partner/parakarry.c [hunk 1]
```diff
@@ -433,7 +434,150 @@ ApiStatus func_80238E24_700BA4(Evt* evt, s32 isInitialCall) {
...
+
+    if (isInitialCall) {
+        script->functionTemp[0] = 0;
+    }
+
+    switch (script->functionTemp[0]) {
+        case 0:
+            state->currentPos.x = partner->currentPos.x;
+            state->currentPos.y = partner->currentPos.y;
+            state->currentPos.z = partner->currentPos.z;
```
Comment (nanaian) [inline src/battle/partner/parakarry.c:449]: Does a struct copy work `state->currentPos = partner->currentPos`?

File: src/battle/partner/parakarry.c [hunk 2]
```diff
@@ -433,7 +434,150 @@ ApiStatus func_80238E24_700BA4(Evt* evt, s32 isInitialCall) {
...
+                    break;
+                }
+
+                do {
+                    if (state->angle >= 0.0f && state->angle < 90.0f) {
+                        state->angle = 270.0f + (270.0f - clamp_angle(state->angle + 180.0f));
+                    } else if (state->angle >= 90.0f && state->angle < 180.0f) {
+                        state->angle = 270.0f - (clamp_angle(state->angle + 180.0f) - 270.0f);
+                    }
+                } while (0);
```
Comment (nanaian) [inline src/battle/partner/parakarry.c:499]: mysterious inlined func?

File: src/battle/partner/parakarry.c [hunk 3]
```diff
@@ -433,7 +434,150 @@ ApiStatus func_80238E24_700BA4(Evt* evt, s32 isInitialCall) {
...
+                    break;
+                }
+
+                do {
+                    if (state->angle >= 90.0f && state->angle < 180.0f) {
+                        state->angle = 360.0f + (360.0f - clamp_angle(state->angle + 180.0f));
+                    } else if (state->angle >= 180.0f && state->angle < 270.0f) {
+                        state->angle = 360.0f - (clamp_angle(state->angle + 180.0f) - 360.0f);
+                    }
+                } while (0); // TODO macro?
```
Comment (nanaian) [inline src/battle/partner/parakarry.c:543]: yes pls / inline func

File: src/battle/partner/parakarry.c [hunk 4]
```diff
@@ -433,7 +434,150 @@ ApiStatus func_80238E24_700BA4(Evt* evt, s32 isInitialCall) {
...
+            state->moveTime = 5;
+            script->functionTemp[0] = 3;
+            D_8023BD70->flags |= 0x10;
+        case 3:
+            add_vec2D_polar(&state->currentPos.x, &state->currentPos.y, state->speed, state->angle);
+            if (state->moveTime == 0) {
+                partner->rotation.z = 0.0f;
+                return ApiStatus_DONE2;
+            }
+            state->moveTime--;
```
Comment (nanaian) [inline src/battle/partner/parakarry.c:573]: Add fallthrough comment

File: src/battle/partner/sushie.c [hunk 5]
```diff
@@ -129,7 +176,178 @@ ApiStatus func_80238480_708120(Evt* script, s32 isInitialCall) {
...
+
+                do {
+                    if (state->angle >= 0.0f && state->angle < 90.0f) {
+                        state->angle = 270.0f + (270.0f - clamp_angle(state->angle + 180.0f));
+                        state->bounceDivisor = rand_int(4) - 2;
+                    } else if (state->angle >= 90.0f && state->angle < 180.0f) {
+                        state->angle = 270.0f - (clamp_angle(state->angle + 180.0f) - 270.0f);
+                        state->bounceDivisor = rand_int(4) - 2;
+                    }
+                } while (0);
```
Comment (nanaian) [inline src/battle/partner/sushie.c:236]: "

File: src/battle/partner/sushie.c [hunk 6]
```diff
@@ -129,7 +176,178 @@ ApiStatus func_80238480_708120(Evt* script, s32 isInitialCall) {
...
+
+                do {
+                    if (state->angle >= 270.0f && state->angle < 360.0f) {
+                        state->angle = 180.0f + (180.0f - clamp_angle(state->angle + 180.0f));
+                        state->bounceDivisor = rand_int(4) - 2;
+                    } else if (state->angle >= 0.0f && state->angle < 90.0f) {
+                        state->angle = 180.0f - (clamp_angle(state->angle + 180.0f) - 180.0f);
+                        state->bounceDivisor = rand_int(4) - 2;
+                    }
+                } while (0); // TODO macro?
```
Comment (nanaian) [inline src/battle/partner/sushie.c:260]: "

**Review by nanaian — CHANGES_REQUESTED** [review]
please make that macro (or preferably inline func!) :pray: :shipit:

**Pull request merged**